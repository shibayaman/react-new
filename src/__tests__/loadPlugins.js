const { loadAllPlugins } = require('../loadPlugins');
const loadPlugins = require('../loadPlugins');

describe('loadAllPlugins', () => {
  it('should load selected plugins', () => {
    const preference = { eslint: true, prettier: true, typescript: true };
    let configKeys = Object.keys(loadAllPlugins(preference).configs);
    expect(configKeys).toEqual(['eslintrc', 'prettierrc', 'tsconfig']);

    preference.eslint = false;
    configKeys = Object.keys(loadAllPlugins(preference).configs);
    expect(configKeys).toEqual(['prettierrc', 'tsconfig']);

    preference.prettier = false;
    preference.typescript = false;
    expect(loadAllPlugins(preference).configs).toEqual({});
  });
});

describe('loadPlugins integration test', () => {
  describe('eslint', () => {
    it('should return unedited eslintrc when no other plugins loaded', () => {
      const preference = { eslint: true, prettier: false, typescript: false };
      expect(loadPlugins(preference).configs.eslintrc).toMatchSnapshot();
    });

    it('should modify eslintrc properly when prettier loaded', () => {
      const preference = { eslint: true, prettier: true, typescript: false };
      expect(loadPlugins(preference).configs.eslintrc).toMatchSnapshot();
    });

    it('should modify eslintrc properly when prettier & ts loaded', () => {
      const preference = { eslint: true, prettier: true, typescript: true };
      expect(loadPlugins(preference).configs.eslintrc).toMatchSnapshot();
    });
  });

  describe('babel config', () => {
    it('should not add babelConfig to configsToTransoform when ts not loaded', () => {
      const preference = { eslint: true, prettier: true, typescript: false };
      expect(
        loadPlugins(preference).configsToTransform.babelConfig
      ).toBeUndefined();
    });

    it('should modify babel config properly when ts loaded', () => {
      const preference = { eslint: true, prettier: true, typescript: true };
      expect(
        loadPlugins(preference).configsToTransform.babelConfig
      ).toMatchSnapshot();
    });
  });

  describe('webpack config', () => {
    it('should not add webpackConfig to configsToTransoform when ts not loaded', () => {
      const preference = { eslint: true, prettier: true, typescript: false };
      expect(
        loadPlugins(preference).configsToTransform.webpackConfig
      ).toBeUndefined();
    });

    it('should modify webpack config properly when ts loaded', () => {
      const preference = { eslint: true, prettier: true, typescript: true };
      expect(
        loadPlugins(preference).configsToTransform.webpackConfig
      ).toMatchSnapshot();
    });
  });
});
