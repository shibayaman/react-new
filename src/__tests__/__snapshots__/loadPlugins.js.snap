// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadPlugins integration test babel config should modify babel config properly when ts loaded 1`] = `
Object {
  "env": Object {
    "test": Object {
      "plugins": Array [
        "transform-es2015-modules-commonjs",
      ],
    },
  },
  "presets": Array [
    Array [
      "@babel/preset-env",
      Object {
        "modules": false,
      },
    ],
    "@babel/preset-react",
    "@babel/preset-typescript",
    "@babel/preset-typescript",
  ],
}
`;

exports[`loadPlugins integration test eslint should modify eslintrc properly when prettier & ts loaded 1`] = `
Object {
  "content": Object {
    "env": Object {
      "browser": true,
      "es6": true,
      "node": true,
    },
    "extends": Array [
      "eslint:recommended",
      "plugin:react/recommended",
      "prettier",
      "prettier/react",
      "prettier",
      "prettier/react",
    ],
    "overrides": Array [
      Object {
        "extends": Array [
          "plugin:@typescript-eslint/eslint-recommended",
          "plugin:@typescript-eslint/recommended",
          "prettier/@typescript-eslint",
        ],
        "files": Array [
          "*.+(ts|tsx)",
        ],
      },
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": Object {
      "ecmaFeatures": Object {
        "jsx": true,
      },
      "ecmaVersion": 2018,
      "sourceType": "module",
    },
    "plugins": Array [
      "react",
      "@typescript-eslint",
    ],
    "settings": Object {
      "react": Object {
        "version": "detect",
      },
    },
  },
  "filename": ".eslintrc",
}
`;

exports[`loadPlugins integration test eslint should modify eslintrc properly when prettier loaded 1`] = `
Object {
  "content": Object {
    "env": Object {
      "browser": true,
      "es6": true,
      "node": true,
    },
    "extends": Array [
      "eslint:recommended",
      "plugin:react/recommended",
      "prettier",
      "prettier/react",
    ],
    "parserOptions": Object {
      "ecmaFeatures": Object {
        "jsx": true,
      },
      "ecmaVersion": 2018,
      "sourceType": "module",
    },
    "plugins": Array [
      "react",
    ],
    "settings": Object {
      "react": Object {
        "version": "detect",
      },
    },
  },
  "filename": ".eslintrc",
}
`;

exports[`loadPlugins integration test eslint should return unedited eslintrc when no other plugins loaded 1`] = `
Object {
  "content": Object {
    "env": Object {
      "browser": true,
      "es6": true,
      "node": true,
    },
    "extends": Array [
      "eslint:recommended",
      "plugin:react/recommended",
    ],
    "parserOptions": Object {
      "ecmaFeatures": Object {
        "jsx": true,
      },
      "ecmaVersion": 2018,
      "sourceType": "module",
    },
    "plugins": Array [
      "react",
    ],
    "settings": Object {
      "react": Object {
        "version": "detect",
      },
    },
  },
  "filename": ".eslintrc",
}
`;

exports[`loadPlugins integration test webpack config should modify webpack config properly when ts loaded 1`] = `
"const path = require('path');
module.exports = env => ({
  mode: env.NODE_ENV || 'production',
  entry: './src/App.tsx',
  output: {
    filename: 'bundle.js',
    path: path.resolve(__dirname, 'build')
  },
  resolve: {
    extensions: [
      '.js',
      '.jsx',
      '.ts',
      '.tsx'
    ]
  },
  module: {
    rules: [{
        test: /.(js|jsx|ts|tsx)$/,
        use: 'babel-loader',
        exclude: /node_modules/
      }]
  },
  devServer: {
    contentBase: path.resolve(__dirname, 'public'),
    compress: true,
    port: 8000
  }
});"
`;
